tests/AsyncResourcePoolTests.cs

```diff
@@ -45,6 +45,156 @@ public async Task ShouldCreateResourcesWhenMinNumResourcesIsZero()
             }
         }
 
+        [Fact(Timeout = Timeout)]
+        public async Task TryGetExisting_ShouldReturnTrue_WhenResourceExists()
+        {
+            var testHarness = new TestHarness();
+
+            const int minNumResources = 1;
+            const int maxNumResources = 1;
+
+            using (var sut = CreateSut(testHarness, minNumResources, maxNumResources))
+            {
+                await Task.Delay(100); // Allow time for resource to be created
```

Comment:
I agree in principle, it just adds a bunch of complexity. I think some of the tests do have this behaviour where we listen for a resource created even and set a `TaskCompletionSource`, feel free to make a PR to improve any of the tests.

I think I'd prefer to keep the tests simpler with a `Task.Delay(100);` rather than spin, might as well do it properly if you're gonna change it. I only run these tests on a quiet machine anyway.